node/src/components/consensus/era_supervisor.rs

```diff
@@ -402,7 +396,7 @@ where
             trace!("executed block in old era {}", block_header.era_id().0);
             return effects;
         }
-        if block_header.switch_block() {
+        if block_header.era_end().is_some() {
```

Comment:
Maybe it wouldn't be a bad idea to keep a `switch_block()` function (which would just return `era_end.is_some()`)? That would be admittedly a _very_ thin wrapper, but I think it is a nice, well-defined abstraction (no matter how we mark a switch block, we're going to have a notion of a switch block and we'll want to be able to check if a block is one).